internal/handlers.go

```diff
@@ -0,0 +1,94 @@
+package internal
+
+import (
+	"encoding/json"
+	"io"
+	"net/http"
+
+	"github.com/rs/zerolog/log"
+
+	"github.com/julienschmidt/httprouter"
+)
+
+func Index(w http.ResponseWriter, r *http.Request, _ httprouter.Params) {
+	w.Header().Set("Content-Type", "application/json; charset=UTF8")
+	w.WriteHeader(http.StatusOK)
+	err := json.NewEncoder(w).Encode("Welcome to GoPay!")
+
+	if err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+		json.NewEncoder(w).Encode(&JsonErrorResponse{Error: &ApiError{Status: http.StatusInternalServerError, Title: "Internal Server Error"}})
+		return
+	}
+}
+
+func GetAllAccounts(w http.ResponseWriter, r *http.Request, _ httprouter.Params) {
+	accs := []*Account{}
+
+	for _, account := range Accounts {
+		accs = append(accs, account)
+	}
+
+	w.Header().Set("Content-Type", "application/json; charset=UTF8")
```

Comment:
Just for context another way to do it is:

```go
	result, err := someService.GetAllAccounts(ctx)
	if err != nil {
		logger.Error(err.Error(), err)
		response.ErrorWithMessage(w, http.StatusInternalServerError, "internal-error", err.Error())
		return
	}

	res, err := jsoniter.Marshal(result)
	if err != nil {
		logger.Error(err.Error(), err)
		response.ErrorWithMessage(w, http.StatusInternalServerError, "internal-error", err.Error())
		return
	}
	response.WithPayload(w, http.StatusOK, res)
```

The response is another package that contains this function to make the code more expressive.

```go
func ErrorWithMessage(w http.ResponseWriter, responseCode int, errorCode, message string) {
		upperErrorCode := strings.ToUpper(errorCode)
	resp := ErrorResponse{
		Error:          upperErrorCode,
		ErrorCode:      upperErrorCode,
		Message:        message,
		UUID:           uuid,
	}

	payload, err := json.Marshal(resp)
	if err != nil {
		WithPayload(w, http.StatusUnprocessableEntity, []byte(`{"error": "Unable to Return Payload"}`))
	}
	WithPayload(w, responseCode, payload)
}

func WithPayload(w http.ResponseWriter, responseCode int, payload []byte) {
	w.Header().Set("Content-Type", "application/json")
	w.WriteHeader(responseCode)
	_, _ = w.Write(payload)
}


```